### PR TITLE
OpInfo for `nn.functional.relu`

### PR DESCRIPTION
**Pull request by krshrimali**
See https://github.com/facebookresearch/functorch/issues/78 and https://github.com/pytorch/pytorch/issues/54261.

cc: @mruberry

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -1032,6 +1032,22 @@ def sample_inputs_linalg_norm(op_info, device, dtype, requires_grad):
...
 
 
+def sample_inputs_nn_activation_relu(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = (
+        (()),
+        ((S, )),
+        ((S, S)),
+        ((S, M, S))
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1042]: We probably don't even need this case

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -6310,6 +6326,13 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypes=all_types_and(torch.float16, torch.bfloat16, torch.bool),
            supports_forward_ad=True,
            sample_inputs_func=sample_inputs_max_min_binary,),
+    OpInfo('nn.functional.relu',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:6329]: Nice!

**Review by mruberry — APPROVED** [review]
Awesome! Thanks @krshrimali.

@zou3519 would you like to check this, make sure it has the sample inputs you need, and shepherd it through?

**Review by zou3519 — APPROVED** [review]

**Comment by zou3519** [comment]
Yup, can do!

**Pull request closed without merging**